### PR TITLE
Base of color-detection tests

### PR DESCRIPTION
**Pull request by Maheliusz**
Basis of color-detection tests, with some color-detection improvements

### CODE REVIEW

File: mrc/localization/color/analyser.py [hunk 1]
```diff
@@ -13,14 +13,12 @@ def __init__(self):
         # TODO: Change predicate to fuzzy the hue on hsv circle
 
     def analyse_chunk(self, image_chunk: ndarray) -> dict:
-        pixel_list = image_chunk.reshape(image_chunk.shape[0] * image_chunk.shape[1], 3)
+        chunk_shape = image_chunk.shape
```
Comment (zdzarsky) [inline mrc/localization/color/analyser.py]: This shall come on the level of review. Please unstage those changes I'll change them in the branch I am pushing right now. img_chunk.shape ~ O(1). I don't feel like it's necessary to add a var here.

File: mrc/localization/color/analyser.py [hunk 2]
```diff
@@ -13,14 +13,12 @@ def __init__(self):
...
+        pixel_list = image_chunk.reshape(chunk_shape[0] * chunk_shape[1], 3)
         cluster = KMeans(n_clusters=self._clusters)
         cluster.fit(pixel_list)
         return self._color_analysis(hsv_pixel_from_centroid(cluster.cluster_centers_))
 
     def _color_analysis(self, hsv_pixel) -> dict:
         (hue, _, _) = hsv_pixel
-        result = {}
```
Comment (zdzarsky) [inline mrc/localization/color/analyser.py]: Plz unstage to avoid mess.

File: mrc/localization/color/extractor.py [hunk 3]
```diff
@@ -15,10 +15,7 @@ def __init__(self, image, blob_coordinates):
         self.analysis = None
 
     def extract(self) -> List[ColorValue]:
-        results = []
-        for chunk in self._chop_blobs():
-            results.append(self._execute_analysis(chunk))
```
Comment (zdzarsky) [inline mrc/localization/color/extractor.py]: plz unstage to avoid mess

File: tests/localization/color/analyser_test.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import unittest
+
+from mrc.localization.color.analyser import Analyser
+
+
```
Comment (zdzarsky) [inline tests/localization/color/analyser_test.py]: We have to add here a new - higher level of abstraction called K-Means strategy.

**Comment by Maheliusz** [comment]
@zdzarsky has a point, those changes in this stage are unnecessary and only make things complicated. Branch closed and its changes will be proposed in later stages of development.

**Pull request closed without merging**